File: php/src/CacheItem.php
<?php

namespace LeastRecentlyUsed;

class CacheItem {
	public $key;
	public $value;
	public $prev;
	public $next;

	public function __construct(string $key, mixed $value) {
		$this->key   = $key;
		$this->value = $value;
		$this->prev  = null;
		$this->next  = null;
	}
}

File: go/lrucache/cacheitem.go
package lrucache

// CacheItem represents an item in the doubly-linked list
type CacheItem struct {
	key   string
	value interface{}
	prev  *CacheItem
	next  *CacheItem
}

// NewCacheItem creates a new cache item with the given key and value
func NewCacheItem(key string, value interface{}) *CacheItem {
	return &CacheItem{
		key:   key,
		value: value,
	}
}
